File: chrome/browser/ui/browser_window/internal/android/browser_window_interface_iterator_android.cc
FASTVC-BENCH-CORPUS:chromium-main-v1-943b94ae-1c74-4335-94fa-ceb4d277cea8
// Copyright 2025 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "chrome/browser/ui/browser_window/public/browser_window_interface_iterator.h"

#include "chrome/browser/ui/browser_window/internal/android/android_browser_window.h"

std::vector<BrowserWindowInterface*> GetAllBrowserWindowInterfaces() {
  return AndroidBrowserWindow::GetAllAndroidBrowserWindowsByCreationTime();
}

std::vector<BrowserWindowInterface*>
GetBrowserWindowInterfacesOrderedByActivation() {
  // TODO(https://crbug.com/419057482): This is wrong, since this is creation
  // order, rather than activation order. This is a temporary solution so things
  // don't crash and "mostly" work (especially when creation order matches
  // activation order, such in the case of a single window).
  return AndroidBrowserWindow::GetAllAndroidBrowserWindowsByCreationTime();
}

BrowserWindowInterface* GetLastActiveBrowserWindowInterfaceWithAnyProfile() {
  std::vector<BrowserWindowInterface*> all_windows =
      GetBrowserWindowInterfacesOrderedByActivation();
  return all_windows.empty() ? nullptr : all_windows[0];
}

File: chrome/browser/ui/browser_window/internal/android/java/src/org/chromium/chrome/browser/ui/browser_window/BrowserWindowInterfaceIteratorAndroid.java
FASTVC-BENCH-CORPUS:chromium-main-v1-943b94ae-1c74-4335-94fa-ceb4d277cea8
// Copyright 2025 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.chrome.browser.ui.browser_window;

import org.jni_zero.CalledByNative;
import org.jni_zero.JniType;

import org.chromium.build.annotations.NullMarked;

/**
 * Java class implementing the native functions in {@code
 * browser_window_interface_iterator_android.cc}.
 */
@NullMarked
final class BrowserWindowInterfaceIteratorAndroid {
    private BrowserWindowInterfaceIteratorAndroid() {}

    @CalledByNative
    @JniType("std::vector<int64_t>")
    private static long[] getAllBrowserWindowInterfaces() {
        return ChromeAndroidTaskTrackerImpl.getInstance().getAllNativeBrowserWindowPtrs();
    }
}
